include/database.py

```diff
@@ -36,18 +36,127 @@ def select(connectionString, sql, parameters):
+ #       raise DatabaseError
+
+ #   except psycopg2.DatabaseError as err:
+ #       sys.stderr.write(f'A general database error occurred: {err}')
+ #       raise DatabaseError
+
+
+#def delete(connectionString, sql, parameters):
+#    try:
+        # Open the connection, override any timeout provided with something shorter
+        # since we expect to be running interactively
+#        connection = psycopg2.connect(connectionString, connect_timeout=1)
+#        cursor = connection.cursor()
+
+        # Execute the query, note the rows
+#        cursor.execute(sql, (parameters['id'],))
+
+#        connection.commit()
+        # check if the study exists
+        #if cursor.rowcount:
+        #    print("Study deleted successfully")
+        #else:
+        #    print("Study does not exist")
+
+        # Clean-up and return
+#        cursor.close()
+#        return cursor.rowcount
+
+
+#    except psycopg2.OperationalError as err:
+#        sys.stderr.write(f'An error occurred connecting to the database: {err}')
+#        raise DatabaseError
+
+#    except psycopg2.DatabaseError as err:
+#        sys.stderr.write(f'A general database error occurred: {err}')
+#        raise DatabaseError
+
+#    except psycopg2.Error as err:
+#        sys.stderr.write(f'An error occurred: {type(err)} {err}')
+#        raise DatabaseError
+
+
+#def update(connectionString, sql, parameters):
+#    try:
+        # Open the connection, override any timeout provided with something shorter
+        # since we expect to be running interactively
+#        connection = psycopg2.connect(connectionString, connect_timeout=1)
+#        cursor = connection.cursor()
+
+        # Execute the query, note the rows
+#        cursor.execute(sql, (parameters['name'], parameters['id'],))
+
+#        connection.commit()
+        # check if the study exists
+#        if cursor.rowcount:
+#            print("Study renamed successfully")
+#        else:
+#            print("Study does not exist")
+
+        # Clean-up and return
+#        cursor.close()
+
+#    except psycopg2.OperationalError as err:
+#        sys.stderr.write(f'An error occurred connecting to the database: {err}')
+#        raise DatabaseError
+
+#    except psycopg2.DatabaseError as err:
+#        sys.stderr.write(f'A general database error occurred: {err}')
+#        raise DatabaseError
+
+#    except psycopg2.Error as err:
+#        sys.stderr.write(f'An error occurred: {type(err)} {err}')
+#        raise DatabaseError
+
+
+def operation(flag, connectionString, sql, parameters):
     try:
         # Open the connection, override any timeout provided with something shorter
         # since we expect to be running interactively
         connection = psycopg2.connect(connectionString, connect_timeout=1)
         cursor = connection.cursor()
+        returnValue = ""
+        if (flag == 1):  # insert
+            cursor.execute(sql, (parameters['Name'],))
```

Comment:
Again, you shouldn't need to do anything with the contents of `parameters` here. Remember the [separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns) here: the call is responsible for supplying a valid SQL query as well as any parameters that are needed for that query, this function is responsible for knowing how to connect to the database server, invoking the SQL provided, and managing the results of that operation (e.g., exceptions).